### DISCUSSION TITLE
Multi-UE and Multi-gNB donwlink problem with USRPs B210

### DISCUSSION THREAD

**Discussion by mohammadih** [srsran/srsRAN_Project#622] [General Help] [answered]
I appreciate if anyone can help with this issue, for 3UE, 3gNBs in uplink iperf3 is working fine, but for downlink, although the pinging for gNB1-->UE1, gNB2 --> UE2 and gNB3 --> UE3 establishes the connection, but when I run the command iperf3 in reverse mode for packet transferring it releasses the connection as you can see in screenshots. Both UE and gNB sides plus the config files for both UE and gNBs are uploaded as well.
[ue log file.zip](https://github.com/srsran/srsRAN_Project/files/15358318/ue.log.file.zip)
[iperf -R error (gnb).zip](https://github.com/srsran/srsRAN_Project/files/15358293/iperf.-R.error.gnb.zip)
[iperf -R (ue).zip](https://github.com/srsran/srsRAN_Project/files/15358294/iperf.-R.ue.zip)
[config_files.zip](https://github.com/srsran/srsRAN_Project/files/15358301/config_files.zip)

**Comment by pgawlowicz** [comment]
Hi, do you run all three gNBs at the same frequency? then probably there is an issue of interference between them. Note that UL SNIR is quite low.

**Reply by mohammadih** [reply]
yes, they are at the same frequency; however, we did the test like this UE1 --> gNB1 --> UE2, again in the downlink the UE2 released. Moreover, for now and simplicity to avoid interference UEs and gNBs are connected through wire.

**Reply by pgawlowicz** [reply]
@mohammadih any update on this issue?

**Reply by mohammadih** [reply]
actually we decided to split gNB side, therefore, each gNB is setup on one machine, but the problem is routing table, in our case we have three paths as follow:
UE1 -->gNB1 -->UE4
UE2 -->gNB2-->UE5
UE3 -->gNB3-->UE6
UE1, UE2 and UE3 are on machine 1
gNB1 on machine 2
gNB2 on machine 3
gNB3 on machine 4
and UE4, UE5 and UE6 on machine 5
now among these three paths always just one path as end to end communication can work, the rest not
eah UE and gNB has its own IP addresses, but what I found, was that when we have end-to-end communication the IP in the host machine (gNB side) that redirect the packets to destination for all 3 distinct gNBs machines was 10.45.1.1; however I defined the IP addresses 10.53.1.1, 10.53.1.4 and 10.53.1.6 for each gNB.
Moreover, I need to mention that each UE and gNB get connected and assigned IPs without problem.
If you want I will provide you the routing table on each machine
I appreciate it if you can help with this issue

**Reply by pgawlowicz** [reply]
do you use the dockerized Open5gs or one manually installed? 

Did all UEs attach to their gnbs?

**Reply by mohammadih** [reply, marked as answer]
yes, problem solved. It was the IP subnet issue.

Thanks
